File: redsheet-portal/src/app/database-models/schedule-email.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class ScheduleEmail {
    constructor(scheduleEmail?: ScheduleEmail )  {
        let properties = [ "emailFrom", "emailTemplateName", "emailTo", "inactivityDays", "isSentScheduleEmail", "moduleName", "projectId", "projectName", "requestedDateTime", "scheduleEmailId",];
        for (let property of properties)
            if (scheduleEmail && scheduleEmail[property])
                this[property] = scheduleEmail[property];
    }
 
    @required()
    @maxLength(200)
	emailFrom : string =   undefined;
 
    @required()
    @maxLength(200)
	emailTemplateName : string =   undefined;
 
    @required()
    @maxLength(200)
	emailTo : string =   undefined;
 
    @range(1,2147483647)
	inactivityDays : number =   undefined;
 
	isSentScheduleEmail : boolean = false ;
 
    @maxLength(1000)
	moduleName : string =   undefined;
 
    @range(1,2147483647)
	projectId : number =   undefined;
 
    @required()
    @maxLength(1000)
	projectName : string =   undefined;
 
    @required()
	requestedDateTime : Date =   undefined;
 
	scheduleEmailId : number =   0 ;


}

File: RedSheet.DbEntities/Models/ScheduleEmail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ScheduleEmails", Schema = "dbo")]
	[Serializable]
	public partial class ScheduleEmail  
	{
 
	[Required] 
	[MaxLength(200)]
	public string EmailFrom { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string EmailTemplateName { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string EmailTo { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int InactivityDays { get; set; }

    [Range(1, int.MaxValue)]
    public int ProjectModuleId { get; set; }

    [Required]
	public bool IsSentScheduleEmail { get; set; } 
 
	[MaxLength(1000)]
	public string ModuleName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string ProjectName { get; set; } 
 
	[Required]
	public System.DateTime RequestedDateTime { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ScheduleEmailId { get; set; } 

	public ScheduleEmail()
	{
    }

    }
}

